src/bank.rs

```diff
@@ -1706,15 +1706,15 @@ mod tests {
         //            let (_, _) = bank.process_ledger(ledger).unwrap();
         //        }
 
-        let window_size = WINDOW_SIZE as usize;
+        let window_size = 128;
         for entry_count in window_size - 3..window_size + 2 {
             let (ledger, pubkey) = create_sample_ledger(entry_count);
             let bank = Bank::default();
             let (tick_height, ledger_height, tail) = bank.process_ledger(ledger).unwrap();
             assert_eq!(bank.get_balance(&pubkey), 1);
             assert_eq!(ledger_height, entry_count as u64 + 3);
             assert_eq!(tick_height, 2);
-            assert!(tail.len() <= window_size);
+            assert!(tail.len() <= WINDOW_SIZE as usize);
```

Comment:
Did this line need to change?

Reply:
yes

Reply:
tail is only rotated at WINDOW_SIZE, so it can grow past artificially limited `window_size`

https://github.com/solana-labs/solana/blob/7ab65352bea4d881269cc43e8bb1dde2972dc30a/src/bank.rs#L1083

Reply:
actually this does kind of mess up this test, I read the range wrong. we could avoid hard-coding window size in this function.

Reply:
looks like @rob-solana is already removing this test anyway in a PR, so maybe we leave it until he gets back.